src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -199,17 +203,14 @@ std::vector<uint_t> calc_new_indices(const reg_t &indices) {
 }
 
 cmatrix_t mul_matrix_by_lambda(const cmatrix_t &mat,
-			       const rvector_t &lambda)
+			       const rvector_t &lambda,
+			       uint_t omp_threads)
 {
   if (lambda == rvector_t {1.0}) return mat;
   cmatrix_t res_mat(mat);
   uint_t num_rows = mat.GetRows(), num_cols = mat.GetColumns();
 
-  #ifdef _WIN32
-     #pragma omp parallel for
-  #else
-     #pragma omp parallel for collapse(2)
-  #endif
+#pragma omp parallel for collapse(2) if (num_rows*num_cols > 64 && MPS::get_omp_threads() > 1) num_threads(MPS::get_omp_threads()) 
```

Comment:
Ok,  you have removed the `#ifdef _WIN32` conditinal define, but this is needed unfortunately.
MSVC compiler doesn't support the  same version of OpenMP that the rest of supported compilers, so we still need it.
I'd suggest to rewrite this like:
```c++
#ifdef _WIN32
   #pragma omp parallel for if (num_rows*num_cols > 64 && MPS::get_omp_threads() > 1) num_threads(MPS::get_omp_threads()) 
#else
   #pragma omp parallel for collapse(2) if (num_rows*num_cols > 64 && MPS::get_omp_threads() > 1) num_threads(MPS::get_omp_threads()) 
#endif
```

Reply:
done